### PR TITLE
Fixed - StackOverflowError in RenewalTask.renewSlots() with many locks #6962

### PR DESCRIPTION
**Pull request by subbotinkv**
## Summary

Fixes [#6962](https://github.com/redisson/redisson/issues/6962) - StackOverflowError in `RenewalTask.renewSlots()` when many locks accumulate.

## Root Cause

Java's `CompletableFuture.thenCompose()` has no tail-call optimization. When futures complete synchronously (common with cached responses or fast Redis), each recursive call adds a stack frame. With enough locks in the renewal map, the stack is exhausted.

## Solution

Replace recursive `thenCompose()` with a **trampoline pattern**:

1. Use `while(true)` loop for synchronous completions (no stack growth)
2. Check `isDone()` before registering callbacks
3. If sync: process in loop (same stack frame)
4. If async: register `whenComplete` callback and return (breaks stack chain)

This approach is better than `thenComposeAsync()` because:
- No thread pool overhead when futures complete synchronously
- Same thread continues when sync, executor only used when truly async
- No risk of task queue exhaustion

## Changes

- Add `AsyncIteratorUtils` for stack-safe async iteration over slots
- Add `AsyncChunkProcessor` for stack-safe chunk processing with callbacks
- Refactor `LockTask`, `ReadLockTask`, `FastMultilockTask` to use `AsyncChunkProcessor`
- Add unit tests including 100,000 element test for stack overflow verification

## Testing

- [x] Unit tests pass (`AsyncChunkProcessorTest`)
- [x] Existing lock tests pass (`RedissonLockTest`, `RedissonReadWriteLockTest`, `RedissonFasterMultiLockTest`)
- [x] Compilation clean with no warnings

### CODE REVIEW

File: redisson/src/main/java/org/redisson/misc/AsyncChunkProcessor.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+
+/**
+ * Utility for processing chunks from an iterator asynchronously without stack overflow.
+ * Uses trampoline pattern to prevent StackOverflowError with large iterators.
+ *
+ * @author Nikita Koksharov
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/misc/AsyncChunkProcessor.java]: the author is not me here

File: redisson/src/main/java/org/redisson/misc/AsyncIteratorUtils.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Function;
+
+/**
+ * Utility for iterating over elements asynchronously without stack overflow.
+ * Uses trampoline pattern to prevent StackOverflowError with large iterators.
+ *
+ * @author Nikita Koksharov
```
Comment (mrniko) [inline redisson/src/main/java/org/redisson/misc/AsyncIteratorUtils.java]: the author is not me here

**Comment by subbotinkv** [comment]
Build issues and review comments have been addressed:
- Fixed checkstyle violation (inline conditional converted to if-else)
- Fixed license headers (copyright year)
- Fixed `@author` tags in new utility classes

Could you please re-review when you have a chance? Thank you!

**Comment by mrniko** [comment]
Thanks for contribution
